Utils/btagSF.C

```diff
@@ -26,75 +27,134 @@ btagSF(inputString, inputFolder, outputFile, treeName, objectName)
+    reader_fastSim    = new BTagCalibrationStandaloneReader(BTagEntryStandalone::OP_MEDIUM, "central", {"up","down"}); 
+
+    reader_fastSim->load(*calib_fast,BTagEntryStandalone::FLAV_UDSG,"fastsim");
+    reader_fastSim->load(*calib_fast,BTagEntryStandalone::FLAV_B,"fastsim");
+    reader_fastSim->load(*calib_fast,BTagEntryStandalone::FLAV_C,"fastsim");
+
+    f_btag_fast_eff = new TFile("/shome/mschoene/btagSF/btageff__SMS-T1bbbb-T1qqqq_fastsim.root"); // Dominick's fast sim b-tagging efficiencies
+    h_btag_fast_eff_b    = (TH2D*) f_btag_fast_eff->Get("h2_BTaggingEff_csv_med_Eff_b"   );
+    h_btag_fast_eff_c    = (TH2D*) f_btag_fast_eff->Get("h2_BTaggingEff_csv_med_Eff_c"   );
+    h_btag_fast_eff_udsg = (TH2D*) f_btag_fast_eff->Get("h2_BTaggingEff_csv_med_Eff_udsg");
+  }
 
+  ~BTagSFHelper(){
+    delete calib;
+    delete reader_fullSim_heavy;
+    delete reader_fullSim_light;
 
+    delete calib_fast;
+    delete reader_fastSim;
 
 
-void get_SF_btag(float pt, float eta, int mcFlavour, float &SF, float &SFup, float &SFdown, bool isFastSim){
+    delete f_btag_eff;
+    delete h_btag_eff_b;
+    delete h_btag_eff_c;
+    delete h_btag_eff_udsg;
 
-  BTagEntry::JetFlavor flavour = BTagEntry::FLAV_UDSG;
-  if      ( abs(mcFlavour)==5 ) flavour = BTagEntry::FLAV_B;
-  else if ( abs(mcFlavour)==4 ) flavour = BTagEntry::FLAV_C;
+    delete f_btag_fast_eff;
+    delete h_btag_fast_eff_b;
+    delete h_btag_fast_eff_c;
+    delete h_btag_fast_eff_udsg;
+  }
+  
+  void get_SF_btag(float pt, float eta, int mcFlavour, float &SF, float &SFup, float &SFdown, bool isFastSim);
+
+  float getBtagEffFromFile(float pt, float eta, int mcFlavour, bool isFastSim);
   
+  void get_weight_btag(int nobj, float* obj_pt, float* obj_eta, int* obj_mcFlavour, float* obj_btagCSV, float &wtbtag, float &wtbtagUp_heavy, float &wtbtagDown_heavy, float &wtbtagUp_light, float &wtbtagDown_light, bool isFastSim);
+
+private:
+  BTagCalibrationStandalone *calib;
+  BTagCalibrationStandaloneReader *reader_fullSim_heavy;
+  BTagCalibrationStandaloneReader *reader_fullSim_light;
+
+  BTagCalibrationStandalone *calib_fast;
+  BTagCalibrationStandaloneReader *reader_fastSim;
+
+
+  TFile *f_btag_eff;
+  TH2D* h_btag_eff_b;
+  TH2D* h_btag_eff_c;
+  TH2D* h_btag_eff_udsg;
+
+
+  TFile *f_btag_fast_eff;
+  TH2D* h_btag_fast_eff_b;
+  TH2D* h_btag_fast_eff_c;
+  TH2D* h_btag_fast_eff_udsg;
+
+};
+
+
+
+
+
+void BTagSFHelper::get_SF_btag(float pt, float eta, int mcFlavour, float &SF, float &SFup, float &SFdown, bool isFastSim){
+
+  BTagEntryStandalone::JetFlavor flavour = BTagEntryStandalone::FLAV_UDSG;
+  if      ( abs(mcFlavour)==5 ) flavour = BTagEntryStandalone::FLAV_B;
+  else if ( abs(mcFlavour)==4 ) flavour = BTagEntryStandalone::FLAV_C;
+  
+  //BM: these two lines are probably unnecessary now with "eval_auto_bounds" method. To be checked
   float pt_cutoff  = std::max(30. ,std::min(669., double(pt)));

```

Comment:
due to this: https://hypernews.cern.ch/HyperNews/CMS/get/btag/1322/1.html
we have to set the lower bound at either 29.999 or 30.001 (I didn't see a difference between the two options) if we use the eval_auto_bounds function